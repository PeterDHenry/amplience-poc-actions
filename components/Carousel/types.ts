import { HeroProps } from '../Hero/types'

export type CarouselProps = {
  heroBannerList: HeroProps[]
}

export const defaultProps: CarouselProps = {
  heroBannerList: [
    {
      _meta: {
        name: 'Default ride like a pro banner',
        schema: 'https://wiggle.com/hero-banner',
        deliveryId: '18a7eee8-6c7a-4a92-8a76-29eb936679a8',
      },
      background: {
        _meta: {
          schema:
            'http://bigcontent.io/cms/schema/v1/core#/definitions/image-link',
        },
        id: 'ef717f48-99e2-4717-89da-6f906056eae1',
        name: 'CCC-11-5-a',
        endpoint: 'wigglecrcnonprd',
        defaultHost: 'cdn.media.amplience.net',
      },
      textAndCTA: {
        headline: 'Default ride like a pro',
        callToActionURL: 'https://www.wiggle.co.uk/pro-riding-for-summer',
        callToActionText: 'Default shop now',
      },
    },
    {
      _meta: {
        name: 'Default Tailored for women banner',
        schema: 'https://wiggle.com/hero-banner',
        deliveryId: 'ecba6af1-17ce-4799-9d20-146b307af92d',
      },
      background: {
        _meta: {
          schema:
            'http://bigcontent.io/cms/schema/v1/core#/definitions/image-link',
        },
        id: '9c6bd46b-21e2-4720-8be6-3fb4b9be6079',
        name: 'homepageh1d',
        endpoint: 'wigglecrcnonprd',
        defaultHost: 'cdn.media.amplience.net',
      },
      textAndCTA: {
        headline: 'Default Tailored for women',
        strapline: 'Your best-ever sports collection',
        callToActionText: 'Default Shop now',
        callToActionURL: 'https://www.wiggle.co.uk/womens-collection',
      },
    },
    {
      _meta: {
        name: 'Default Discover different banner',
        schema: 'https://wiggle.com/hero-banner',
        deliveryId: 'a6cac415-3411-4481-a741-c7d4e1edbff2',
      },
      background: {
        _meta: {
          schema:
            'http://bigcontent.io/cms/schema/v1/core#/definitions/image-link',
        },
        id: 'a11cb660-fd4a-498f-8222-022edfeb479b',
        name: 'homepage-h-d',
        endpoint: 'wigglecrcnonprd',
        defaultHost: 'cdn.media.amplience.net',
      },
      textAndCTA: {
        headline: 'Default Discover different',
        callToActionText: 'Default Find out more',
        callToActionURL: 'https://www.wiggle.co.uk/discover-different',
      },
    },
  ],
}
