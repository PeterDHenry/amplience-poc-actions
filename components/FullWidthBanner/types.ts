export type FullWidthBannerProps = {
  _meta: {
    name: string
    schema: string
    deliveryId: string
  }
  image: { name: string }
  textAndCTA: {
    callToActionURL: string
    callToActionText: string
    headline: string
    strapline: string
  }
}

export const defaultProps: FullWidthBannerProps = {
  _meta: {
    name: 'string',
    schema: 'string',
    deliveryId: '1543512973',
  },
  image: { name: 'homepageh1d' },
  textAndCTA: {
    callToActionURL: 'https://www.wiggle.co.uk/run-new-season-2021',
    callToActionText: 'Default cta text',
    headline: 'Default header',
    strapline: 'Default subheader',
  },
}
