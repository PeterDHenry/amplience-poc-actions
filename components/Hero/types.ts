export type HeroProps = {
  _meta: {
    name: string
    schema: string
    deliveryId: string
  }
  cssClass?: string
  background: {
    _meta: {
      schema: string
    }
    id: string
    name: string
    endpoint: string
    defaultHost: string
  }
  //color: string
  //contrastColor: string
  textAndCTA: {
    callToActionURL: string
    callToActionText: string
    headline: string
    strapline?: string
  }
}

export const defaultProps: HeroProps = {
  _meta: {
    name: 'Ride like a pro banner',
    schema: 'https://wiggle.com/hero-banner',
    deliveryId: '18a7eee8-6c7a-4a92-8a76-29eb936679a8',
  },
  background: {
    _meta: {
      schema: 'http://bigcontent.io/cms/schema/v1/core#/definitions/image-link',
    },
    id: 'ef717f48-99e2-4717-89da-6f906056eae1',
    name: 'CCC-11-5-a',
    endpoint: 'wigglecrcnonprd',
    defaultHost: 'cdn.media.amplience.net',
  },
  textAndCTA: {
    headline: 'Ride like a pro',
    callToActionURL: 'https://www.wiggle.co.uk/pro-riding-for-summer',
    callToActionText: 'Shop now',
  },
}
