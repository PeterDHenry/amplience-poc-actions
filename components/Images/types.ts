export type LazyLoadImageProps = {
  altText: string
  animationDuration: number
  applyAspectRatio: boolean
  height: number
  id?: string
  width: number
}

export type HeroImageProps = {
  altText: string
  id: string
}

export const defaultLazyLoadImageProps: LazyLoadImageProps = {
  altText: 'Default alt text',
  id: 'homepage-h-d',
  animationDuration: 0.1,
  width: 1920,
  height: 650,
  applyAspectRatio: true,
}

export const defaultHeroImageProps: HeroImageProps = {
  altText: 'Default hero alt text',
  id: 'homepage-h-d',
}
