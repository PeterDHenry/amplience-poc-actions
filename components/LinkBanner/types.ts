export type LinkBannerProps = {
  textAndCTA: {
    callToActionURL: string
    callToActionText: string
    headline: string
    strapline: string
  }
  backgroundColour: string
  fontColour: string
}

export const defaultLinkBannerProps: LinkBannerProps = {
  textAndCTA: {
    callToActionURL: 'https://www.wiggle.co.uk/sc/wiggle-plus',
    callToActionText: 'Find out more',
    headline:
      'Wiggle+ unlimited next day delivery & FREE RETURNS for a year - only Â£9.99',
    strapline: '',
  },
  backgroundColour: '#F7F7F7',
  fontColour: '#3C3C3B',
}
