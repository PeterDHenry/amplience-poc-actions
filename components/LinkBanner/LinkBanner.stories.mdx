import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks'
import LinkBanner from './LinkBanner'

<Meta
  title="CMS Content/LinkBanner"
  component={LinkBanner}
  argTypes={{
    backgroundColour: {
      options: ['#F7F7F7'],
      control: { type: 'select' },
    },
    fontColour: {
      options: ['#3C3C3B'],
      control: { type: 'select' },
    },
  }}
  args={{
    textAndCTA: {
      callToActionURL: 'https://www.wiggle.co.uk/sc/wiggle-plus',
      callToActionText: 'Find out more',
      headline:
        'Wiggle+ unlimited next day delivery & FREE RETURNS for a year - only Â£9.99',
    },
    backgroundColour: '#F7F7F7',
    fontColour: '#3C3C3B',
  }}
/>

export const Template = (args) => <LinkBanner {...args} />

<Canvas>
  <Story name="Wiggle Plus Banner">{Template.bind({})}</Story>
</Canvas>
