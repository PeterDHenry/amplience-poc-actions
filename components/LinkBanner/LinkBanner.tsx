import React from 'react'
import styled from 'styled-components'
import { LinkBannerProps } from './types'

const Banner = styled.a<any>`
  padding: 10px 5px;
  margin: 10px 0;
  background-color: ${(props) => props.backgroundColour};
  color: ${(props) => props.fontColour};
  text-decoration: none;
  font-family: 'Roboto Condensed';
  display: flex;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
  &:hover > .cta-btn {
    transition: 0.4s;
    -webkit-transition: 0.4s;
    -moz-transition: 0.4s;
    -ms-transition: 0.4s;
    -o-transition: 0.4s;
    border: 2px solid #3c3c3b;
    background: ${(props) => props.fontColour};
    color: #fff;
  }
  &:hover {
    color: ${(props) => props.fontColour};
  }
  p {
    margin: 0;
  }
`

const BannerText = styled.div<any>`
  margin-right: 10px;
  display: flex;
  flex-wrap: nowrap;
  align-items: center;
`

const BannerCTA = styled.button<any>`
  border: 2px solid ${(props) => props.fontColour};
  padding: 2px 16px;
  text-align: center;
  background: none;
  font-size: 16px;
  transition: 0.4s;
  -webkit-transition: 0.4s;
  -moz-transition: 0.4s;
  -ms-transition: 0.4s;
  -o-transition: 0.4s;
`

const LinkBanner = ({
  textAndCTA: { callToActionURL, callToActionText, headline },
  backgroundColour,
  fontColour,
}: LinkBannerProps) => (
  <Banner
    href={callToActionURL}
    backgroundColour={backgroundColour}
    fontColour={fontColour}
    className="link-banner"
  >
    <BannerText>
      <p>{headline.toUpperCase()}</p>
    </BannerText>
    <BannerCTA className="cta-btn">
      <p>{callToActionText.toUpperCase()}</p>
    </BannerCTA>
  </Banner>
)

LinkBanner.defaultProps = {
  textAndCTA: {
    callToActionURL: 'https://www.wiggle.co.uk/',
    callToActionText: 'Test CTA Text',
    headline: 'Test Headline',
    strapline: 'Test Strapline',
  },
  backgroundColour: '#F7F7F7',
  fontColour: '#3C3C3B',
}

export default LinkBanner
