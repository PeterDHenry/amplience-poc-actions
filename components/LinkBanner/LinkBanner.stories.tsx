import { Meta, Story } from '@storybook/react'
import { LinkBanner as Component } from './LinkBanner'
import { defaultLinkBannerProps, LinkBannerProps } from './types'

const meta: Meta = {
  title: 'CMS Content/LinkBanner',
  component: Component,
  argTypes: {
    backgroundColour: {
      options: ['#F7F7F7', '#DBDBDB', '#9E9E9E'],
      control: { type: 'select' },
    },
    fontColour: {
      options: ['#3C3C3B', '#595959', '#363636', '#2E2E2E'],
      control: { type: 'select' },
    },
  },
}

export default meta

type StoryProps = LinkBannerProps

export const LinkBanner: Story<StoryProps> = (props) => <Component {...props} />

LinkBanner.args = defaultLinkBannerProps
