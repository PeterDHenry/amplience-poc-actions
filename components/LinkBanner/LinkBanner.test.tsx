import { render } from '@testing-library/react'
import LinkBanner from './LinkBanner'
import { LinkBannerProps } from './types'

describe('Link Banner component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<LinkBanner {...linkBannerProps} />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})

const linkBannerProps: LinkBannerProps = {
  mainText: 'Default Text',
  ctaText: 'Default CTA Text',
  backgroundColour: '#F7F7F7',
  fontColour: '#000',
  link: 'https://www.wiggle.co.uk/sc/wiggle-plus',
  showOnMobile: true
}
