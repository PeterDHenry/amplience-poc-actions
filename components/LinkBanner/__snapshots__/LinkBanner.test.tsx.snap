// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Banner component Matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="LinkBanner__Banner-sc-1rg8ysw-0 cpXOje link-banner"
    href="https://www.wiggle.co.uk/"
  >
    <div
      class="LinkBanner__BannerText-sc-1rg8ysw-1 bQArJM"
    >
      <p>
        TEST HEADLINE
      </p>
    </div>
    <button
      class="LinkBanner__BannerCTA-sc-1rg8ysw-2 hRGaUq cta-btn"
    >
      <p>
        TEST CTA TEXT
      </p>
    </button>
  </a>
</DocumentFragment>
`;
